rust/tools/authorization-logic/src/parsing/printvisitor.rs

```diff
@@ -0,0 +1,52 @@
+use crate::parsing::authlogicvisitor::*;
+use crate::parsing::authlogicparser::*;
+use antlr_rust::tree::{ParseTree,ParseTreeVisitor};
+
+// this code is now vestigial, but shows roughly how to make a visitor that side-effects.
```

Comment:
nit: Comment should start with a capital letter. Also, it might be better to move this code elsewhere to an example if it demonstrates something useful.